staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -238,6 +240,11 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 		// or supporting retry mechanisms by not sending non-fatal errors to the clients
 		proxy.ErrorHandler = h.Responder.Error
 	}
+	if h.OverrideRequestHost {
+		// exchanging req.Host with the backend location is necessary for backends that act on the HTTP host header (e.g. API gateways),
+		// because req.Host has preference over req.URL.Host in filling this header field
+		newReq.Host = h.Location.Host
```

Comment:
A minor comments here, newReq setters should be inserted after Line 231.
Line 227 - Line 231 for newReq settings.
Line 233 - Line 242 for proxy setings.

> As  req.Host has preference over  req.URL.Host
Why not check whether they are different or newReq.Host is nil before override request host  every time？

Reply:
@pacoxu Done!